solenspipe/solenspipe.py

```diff
@@ -194,28 +298,68 @@ def get_qfunc(px,ucls,mlmax,est1,Al1=None,est2=None,Al2=None,R12=None):
     else:
         bh = False
 
-    assert est1 in ['TT','TE','EE','EB','TB','MV','MVPOL'] # TODO: add other
-    qfunc1 = lambda X,Y: qe.qe_all(px,ucls,mlmax,
-                                   fTalm=Y[0],fEalm=Y[1],fBalm=Y[2],
-                                   estimators=[est1],
-                                   xfTalm=X[0],xfEalm=X[1],xfBalm=X[2])[est1]
+    assert est1 in ['TT','TE','EE','EB','TB','MV','MVPOL','SHEAR'] # TODO: add other
+    if est1=='SHEAR':
+        qfunc1 = lambda X,Y: qe.qe_shear(px,mlmax,
+                            Talm=X[0],fTalm=Y[1])
+    else:
+        qfunc1 = lambda X,Y: qe.qe_all(px,ucls,mlmax,
+                                    fTalm=Y[0],fEalm=Y[1],fBalm=Y[2],
+                                    estimators=[est1],
+                                    xfTalm=X[0],xfEalm=X[1],xfBalm=X[2])[est1]
 
     if bh:
-        assert est2 in ['SRC'] # TODO: add mask
-        qfunc2 = lambda X,Y: qe.qe_pointsources(px,mlmax,fTalm=Y[0],xfTalm=X[0])
+        assert est2 in ['SRC','PH','MASK'] # TODO: add mask
+        if est2=='SRC' and profile is None:
+            qfunc2 = lambda X,Y: qe.qe_pointsources(px,mlmax,fTalm=Y[0],xfTalm=X[0])
+        elif est2=='SRC':
+            print("profile hardening")
+            qfunc2 = lambda X,Y: qe.qe_source(px,mlmax,Y[0],profile=profile,xfTalm=X[0])
+        elif est2=='mask':
+            qfunc2 = lambda X,Y: qe.qe_mask(px,ucls,mlmax,fTalm=Y[0],xfTalm=X[0])
         # The bias-hardened estimator Eq 27 of arxiv:1209.0091
         if R12.shape[0]==1:
-            # Bias harden only gradient e.g. source hardening
-            def retfunc(X,Y):
-                q1 = qfunc1(X,Y)
-                q2 = qfunc2(X,Y)
-                g = cs.almxfl( \
-                               (cs.almxfl(q1[0],Al1[0]) - \
-                                cs.almxfl(qfunc2(X,Y),Al1[0] * Al2 * R12[0])) , \
-                               1. / (1. - Al1[0] * Al2 * R12[0]**2.) \
-                )
-                c = cs.almxfl(q1[1],Al1[1])
-                return np.asarray((g,c))
+
+            if est1=='TT':
+                # Bias harden only gradient e.g. source hardening
+                def retfunc(X,Y):
+                    q1 = qfunc1(X,Y)
+                    q2 = qfunc2(X,Y)
+                    g = cs.almxfl( \
+                                (cs.almxfl(q1[0],Al1[0]) - \
+                                    cs.almxfl(qfunc2(X,Y),Al1[0] * Al2 * R12[0])) , \
+                                1. / (1. - Al1[0] * Al2 * R12[0]**2.) \
+                    )
+                    c = cs.almxfl(q1[1],Al1[1])
+                    return np.asarray((g,c))
+            else:
```

Comment:
The below looks specific to the MV case, but the `else` doesn't check for that specifically. What happens if some other estimator is passed as `est1`?